src/Controllers/SettingsController.php

```diff
@@ -52,6 +53,97 @@ public function __construct(
+    {
+        $user = $this->auth->user();
+
+        config('buildup_start');
+        config('teardown_end');
+
+        $data = $this->validate($request, [
+            'pronoun'                => 'optional|max:15',
+            'first_name'             => 'optional|max:64',
+            'last_name'              => 'optional|max:64',
+            'planned_arrival_date'   => 'required|date:Y-m-d',
+            'planned_departure_date' => 'optional|date:Y-m-d',
+            'dect'                   => 'optional|length:0:40', // dect/mobile can be purely numbers. "max" would have
+            'mobile'                 => 'optional|length:0:40', // checked their values, not their character length.
+            'email'                  => 'required|email|max:254',
+            'email_shiftinfo'        => 'optional|checked',
+            'email_news'             => 'optional|checked',
+            'email_human'            => 'optional|checked',
+            'email_goody'            => 'optional|checked',
+            'shirt_size'             => 'required',
+        ]);
+
+        if (config('enable_pronoun')) {
+            $user->personalData->pronoun = $data['pronoun'];
+        }
+
+        if (config('enable_user_name')) {
+            $user->personalData->first_name = $data['first_name'];
+            $user->personalData->last_name = $data['last_name'];
+        }
+
+        if (config('enable_planned_arrival')) {
+            if (!$this->isArrivalDateValid($data['planned_arrival_date'], $data['planned_departure_date'])) {
+                $this->addNotification('settings.profile.planned_arrival_date.invalid', 'errors');
+                return $this->redirect->to('/settings/profile');
+            } elseif (!$this->isDepartureDateValid($data['planned_arrival_date'], $data['planned_departure_date'])) {
+                $this->addNotification('settings.profile.planned_departure_date.invalid', 'errors');
+                return $this->redirect->to('/settings/profile');
+            } else {
+                $user->personalData->planned_arrival_date = $data['planned_arrival_date'];
+                $user->personalData->planned_departure_date = $data['planned_departure_date'] ?: null;
+            }
+        }
+
+        if (config('enable_dect')) {
+            $user->contact->dect = $data['dect'];
+        }
+
+        $user->contact->mobile = $data['mobile'];
+        $user->email = $data['email'];
+        $user->settings->email_shiftinfo = $data['email_shiftinfo'] ?: false;
+        $user->settings->email_news = $data['email_news'] ?: false;
+        $user->settings->email_human = $data['email_human'] ?: false;
+
+        if (config('enable_goody')) {
+            $user->settings->email_goody = $data['email_goody'] ?: false;
+        }
+
+        if (isset(config('tshirt_sizes')[$data['shirt_size']])) {
+            $user->personalData->shirt_size = $data['shirt_size'];
+            $user->personalData->save();
```

Comment:
Really? Where?
Or do you mean the _UserShirtController_? Because the only duplication I see there are these 4 lines.
In the UserShirtController, I could make these lines a public function and let the SettingsController call it. I didn't do it because I didn't see the huge benefit in it in this case. But I'm open for suggestions.

Reply:
Or you probably meant just the $user->personalData->save();